### PR TITLE
[object.pick] Simplify types

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
`U[]` is not needed

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by ChocolateLoverRaj** [comment]
The existing tests already test with normal array and readonly array

**Review by MysteryBlokHed — APPROVED** [review]
LGTM, but the PR will still need a maintainer's approval since no tests are modified.

**Review by domdomegg — APPROVED** [review]

**Review by amcasey — APPROVED** [review]

**Comment by ChocolateLoverRaj** [comment]
Ready to merge

**Pull request merged**
